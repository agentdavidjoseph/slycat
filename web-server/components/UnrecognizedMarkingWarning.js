import React from "react";
import server_root from "js/slycat-server-root";

let UnrecognizedMarkingWarning = (params) =>
{
  function showModel()
  {
    // Hide the overlay that is covering up the model
    params.warning_element.setAttribute('class', 'invisible');
  }

  return (
    <div className="card mt-5 mx-5">
      <div className="card-header bg-warning text-center">
        <h4 className="card-title mb-0">Warning</h4>
      </div>
      <div className="card-body bg-light text-center">
        <h5 className="card-title">Unrecognized Marking</h5>
        <p className="card-text">This model has a marking that is not recognize.</p>
        {params.marking &&
          <p><code>{params.marking}</code></p>
        }
        <p className="mb-0">Choose one of the options below.</p>
      </div>
      <ul className="list-group list-group-flush">
        <li className="list-group-item pb-3">
          <h5>Go Back To Project</h5>
          <p>If you do not wish to view this model, you can go back to the project.</p>
          <a href={`${server_root}projects/${params.project_id}`} className="btn btn-primary">Go Back To Project</a>
        </li>
        <li className="list-group-item pb-3">
          <h5>Edit Marking & Display Model</h5>
          <p>
            If you wish to edit this model's marking and then display it:
          </p>

          <ol className="mb-4">
            <li>Click the <strong className="text-nowrap">Edit</strong> menu above.</li>
            <li>Select <strong className="text-nowrap">Model Name & Info</strong>.</li>
            <li>Choose a new marking.</li>
            <li>Click the <strong className="text-nowrap">Save Changes</strong> button.</li>
          </ol>

          <div className="alert alert-warning mb-0" role="alert">
            The model will display immediately after you choose the new marking and click the <strong className="text-nowrap">Save Changes</strong> button.
          </div>
        </li>
        <li className="list-group-item pb-3">
          <h5>Display Model Now</h5>
          <p>If you wish to view this model now, you can display it even with its unrecognized marking.</p>
          <button type="button" className="btn btn-danger" 
            onClick={showModel}>
            Display Model Now
          </button>
        </li>
      </ul>
    </div>
  )
}

export default UnrecognizedMarkingWarning;