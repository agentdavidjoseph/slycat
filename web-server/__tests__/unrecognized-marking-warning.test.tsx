
import React from 'react';
import UnrecognizedMarkingWarning, 
{UnrecognizedMarkingWarningProps, UnrecognizedMarkingWarningState} 
from "components/UnrecognizedMarkingWarning";
import { mount } from "enzyme";


describe('when loading a bad marking the UnrecognizedMarkingWarning',() =>{ 
  const marking="empty";
  const warning_element= {setAttribute:jest.fn()};
  const render = mount(
    <UnrecognizedMarkingWarning
      marking = {marking}
      warning_element = {warning_element}
      project_id = "1234"
    />
  );
  test('we have expected props on initial load', () => {
    const props = render.props() as UnrecognizedMarkingWarningProps;
    expect(props).toMatchSnapshot();
  });

  test('we no state', () => {
    const state = render.state() as UnrecognizedMarkingWarningState;
    expect(state).toMatchSnapshot();
  });

  test('we should have a full initialization', () => {
    expect(render).toMatchSnapshot();
  });

  test('we should have a show modal function that calls setAttribute', () => {
    render.instance().showModel();
    expect(warning_element.setAttribute).toHaveBeenCalled();
  });

});

export default undefined