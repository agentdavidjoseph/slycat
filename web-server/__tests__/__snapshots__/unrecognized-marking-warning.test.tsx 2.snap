// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a bad marking the UnrecognizedMarkingWarning we have expected props on initial load 1`] = `
Object {
  "marking": "empty",
  "project_id": "1234",
  "warning_element": Object {
    "setAttribute": [MockFunction],
  },
}
`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we no state 1`] = `Object {}`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we should have a full initialization 1`] = `
<UnrecognizedMarkingWarning
  marking="empty"
  project_id="1234"
  warning_element={
    Object {
      "setAttribute": [MockFunction],
    }
  }
>
  <div
    className="card mt-5 mx-5"
  >
    <div
      className="card-header bg-warning text-center"
    >
      <h4
        className="card-title mb-0"
      >
        Unrecognized Marking
      </h4>
    </div>
    <div
      className="card-body bg-light text-center"
    >
      <p
        className="card-text"
      >
        This model has a marking that is not recognized.
      </p>
      <p
        className="mb-0"
      >
        <code>
          empty
        </code>
      </p>
    </div>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item pb-3"
      >
        <h5>
          Edit Marking to Display Model
        </h5>
        <p>
          If you wish to edit this model's marking and then display it:
        </p>
        <ol
          className="mb-4"
        >
          <li>
            Click the 
            <strong
              className="text-nowrap"
            >
              Edit
            </strong>
             dropdown button at the top of the page.
          </li>
          <li>
            Select 
            <strong
              className="text-nowrap"
            >
              Model Name & Info
            </strong>
            .
          </li>
          <li>
            Choose a new marking.
          </li>
          <li>
            Click the 
            <strong
              className="text-nowrap"
            >
              Save Changes
            </strong>
             button.
          </li>
        </ol>
        <div
          className="alert alert-warning mb-0"
          role="alert"
        >
          The model will display immediately after you choose the new marking and click the 
          <strong
            className="text-nowrap"
          >
            Save Changes
          </strong>
           button.
        </div>
      </li>
      <li
        className="list-group-item pb-3"
      >
        <a
          href="/projects/1234"
        >
          ‚Üê Go back to project
        </a>
      </li>
    </ul>
  </div>
</UnrecognizedMarkingWarning>
`;
